File: python/shipping_child_workflow.py
from datetime import timedelta
import logging

from temporalio import workflow

from activities import OrderActivities
from shared_objects import OrderInput, OrderItem

logging.basicConfig(level=logging.INFO)


@workflow.defn
class ShippingChildWorkflow:

    def __init__(self):
        self.retry_policy = OrderActivities.retry_policy

    @workflow.run
    async def execute(self, input: OrderInput, item: OrderItem):
        workflow.logger.info("Shipping workflow started, orderId " + input.OrderId)

        await workflow.start_activity_method(
            OrderActivities.ship_order,
            args=[input, item],
            start_to_close_timeout=timedelta(seconds=5),
            retry_policy=self.retry_policy
        )

File: dotnet/src/ShippingChildWorkflow.cs
using DotNetOrderManagement.model;
using Microsoft.Extensions.Logging;
using Temporalio.Workflows;

namespace DotNetOrderManagement;

[Workflow]
public class ShippingChildWorkflow
{
    [WorkflowRun]
    public async Task Execute(OrderInput input, OrderItem orderItem)
    {
        Workflow.Logger.LogInformation("Shipping workflow started, orderId = {}", input.OrderId);
        
        // Ship order
        await Workflow.ExecuteActivityAsync((OrderActivities act) =>
            act.ShipOrder(input, orderItem), OrderActivities.ActivityOpts);
    }
}